src/DssSpell.sol

```diff
@@ -44,237 +40,163 @@ contract DssSpellAction is DssAction {
+        //  Add PSM-GUSD-A as a new Vault Type
+        //  https://vote.makerdao.com/polling/QmayeEjz?network=mainnet#poll-detail
+        DssExecLib.addNewCollateral(
+            CollateralOpts({
+                ilk:                   "PSM-GUSD-A",
+                gem:                   GUSD,
+                join:                  MCD_JOIN_PSM_GUSD_A,
+                clip:                  MCD_CLIP_PSM_GUSD_A,
+                calc:                  MCD_CLIP_CALC_PSM_GUSD_A,
+                pip:                   PIP_GUSD,
+                isLiquidatable:        false,
+                isOSM:                 false,
+                whitelistOSM:          false,
+                ilkDebtCeiling:        10 * MILLION,
+                minVaultAmount:        0,
+                maxLiquidationAmount:  0,
+                liquidationPenalty:    1300,
+                ilkStabilityFee:       ZERO_PCT_RATE,
+                startingPriceFactor:   10500,
+                breakerTolerance:      9500, // Allows for a 5% hourly price drop before disabling liquidations
+                auctionDuration:       220 minutes,
+                permittedDrop:         9000,
+                liquidationRatio:      10000,
+                kprFlatReward:         300,
+                kprPctReward:          10 // 0.1%
+            })
+        );
+        DssExecLib.setStairstepExponentialDecrease(MCD_CLIP_CALC_PSM_GUSD_A, 120 seconds, 9990);
+        DssExecLib.setIlkAutoLineParameters("PSM-GUSD-A", 10 * MILLION, 10 * MILLION, 24 hours);
 
-        // Increase WBTC-A Stability Fee (duty)
-        // https://vote.makerdao.com/polling/QmRUgsvi?network=mainnet#poll-detail
-        // https://forum.makerdao.com/t/mid-month-parameter-changes-proposal-ppg-omc-001-2021-11-10/11562
-        DssExecLib.setIlkStabilityFee("WBTC-A", FOUR_PCT_RATE, true); // From 2.5% to 4%
+        //  Core Unit Budget Distributions
+        DssExecLib.sendPaymentFromSurplusBuffer(SAS_WALLET, 245_738);
+        DssExecLib.sendPaymentFromSurplusBuffer(IS_WALLET, 195_443);
+        DssExecLib.sendPaymentFromSurplusBuffer(DECO_WALLET, 465_625);
 
-        // Increase LRC-A Target Liquidation Ratio (mat)
-        address LRC_LERP = LerpFactoryAbstract(OLD_LERP_FAB).lerps("LRC Offboarding");
-        uint256 CURRENT_LRC_MAT = LerpAbstract(LRC_LERP).tick();
-        SpotAbstract(DssExecLib.spotter()).deny(LRC_LERP);
-        DssExecLib.linearInterpolation({
-            _name:      "LRC-A Offboarding",
-            _target:    DssExecLib.spotter(),
-            _ilk:       "LRC-A",
-            _what:      "mat",
-            _startTime: block.timestamp,
-            _start:     CURRENT_LRC_MAT,
-            _end:       TARGET_LRC_MAT,
-            _duration:  30 days
-        });
+        VestAbstract(MCD_VEST_DAI).restrict(
+            VestAbstract(MCD_VEST_DAI).create(RWF_WALLET, 1_860_000.00 * 10**18, JAN_01_2022, DEC_31_2022 - JAN_01_2022, 0, address(0))
```

Comment:
This is a minimal thing, but if a vest ends on DEC_31_2022 shouldn't be at the end of the day?
So I think this should look like:
```suggestion
            VestAbstract(MCD_VEST_DAI).create(RWF_WALLET, 1_860_000.00 * 10**18, JAN_01_2022, DEC_31_2022 + 1 days - 1 - JAN_01_2022, 0, address(0))
```

Reply:
Anyway up to you, 1 day won't make a practical difference.

Reply:
Yeah technically I think you're right, but precedent seems to indicate we don't care about this: https://github.com/makerdao/spells-mainnet/blob/master/archive/2021-09-03-DssSpell/DssSpell.sol#L119

Reply:
Well I actually took care of it in that spell. As added one extra day in the date itself. So if something had to finish on 31st December, then I put the duration as 1st January - start date.
I didn't do the `- 1`  though.
Anyway I think is fine, we shouldn't delay deploying the spell due this.